src/runtimes/node/bundler.js

```diff
@@ -88,18 +114,40 @@ const bundleJsFile = async function ({
   }
 }
 
-const getCleanupFunction =
-  (...paths) =>
-  async () => {
-    await Promise.all(paths.filter(Boolean).map(safeUnlink))
-  }
+// Takes the `outputs` object produced by esbuild and returns a Map with the
+// absolute paths of the generated files as keys, and the paths that those
+// files should take in the generated bundle as values. This is compatible
+// with the `aliases` format used upstream.
+const getBundlePaths = ({ destFolder, outputs, srcFile }) => {
+  const bundleFilename = `${basename(srcFile, extname(srcFile))}.js`
+  const mainFileDirectory = dirname(srcFile)
+  const bundlePaths = new Map()
 
-const getSourcemapPath = (outputs) => {
-  const relativePath = Object.keys(outputs).find((path) => extname(path) === '.map')
+  // The paths returned by esbuild are relative to the current directory, which
+  // is a problem on Windows if the target directory is in a different drive
+  // letter. To get around that, instead of using `path.resolve`, we compute
+  // the absolute path by joining `destFolder` with the `basename` of each
+  // entry of the `outputs` object.
+  Object.entries(outputs).forEach(([path, output]) => {
+    const filename = basename(path)
+    const extension = extname(path)
+    const absolutePath = join(destFolder, filename)
 
-  if (relativePath) {
-    return resolve(relativePath)
-  }
+    if (output.entryPoint && basename(output.entryPoint) === basename(srcFile)) {
+      // Ensuring the main file has a `.js` extension.
+      const normalizedSrcFile = getPathWithExtension(srcFile, '.js')
+
+      bundlePaths.set(absolutePath, normalizedSrcFile)
+    } else if (extension === '.js' || filename === `${bundleFilename}.map`) {
```

Comment:
This looks for any additional files, as long as they are JS files (e.g. dynamic import shims) or sourcemap files. If a file doesn't fall into this criteria, we don't care about it.